### DISCUSSION TITLE
Cannot view 1 minute rates on a 15 second scrape interval

### DISCUSSION THREAD

**Discussion by braunsonm** [grafana/mimir#9289] [Help and support] [answered]
Hello! I think I might be confused on how a rate query is supposed to work and I haven't found any information online that quite explains it. From what I understand, a rate query requires two samples within the time window you're searching.

My setup is Alloy scraping a target -> remote write to Mimir.

Take the following as an example:
`cortex_request_duration_seconds_count` has a data point every 15 seconds.
`rate(cortex_request_duration_seconds_count[1m])` returns no data.
`rate(cortex_request_duration_seconds_count[2m])` has a point every 15 seconds.

What does my 1 minute rate window not work in this case? I don't remember having this issue in Prometheus before I switched to Mimir with similar rates.

**Comment by pstibrany** [comment, marked as answer]
Hello. To understand why `rate(cortex_request_duration_seconds_count[1m])` returns no results, we would need to see raw samples used for the computation.

You can see raw samples in Grafana by using 1) Instant query type, 2) using query: `cortex_request_duration_seconds_count[$__range]`. (In my example, I also used `{namespace="mimir-dev-12"}` to reduce number of returned series).

<img width="1455" alt="image" src="https://github.com/user-attachments/assets/b013f0bb-cd1e-49b8-ba67-548ce0159c6a">

Given that `rate(cortex_request_duration_seconds_count[2m])` returns data, most likely explanation is that there are not 2 samples in 1m window. (that is, your scrape interval is likely >= 1 minute).

> `cortex_request_duration_seconds_count` has a data point every 15 seconds.
> `rate(cortex_request_duration_seconds_count[2m])` has a point every 15 seconds.

Range query will produce result at every "step" of the requested interval. This just means that your step is 15 seconds. But the fact that rate of 2m windows shows results proves that in every 2m window there are at least two samples it can work with.

**Reply by braunsonm** [reply]
This was a wonderfully detailed answer. Thank you.

> Range query will produce result at every "step" of the requested interval. This just means that your step is 15 seconds. 

This was exactly my misconception, seeing 15 second results made me think I was scraping at 15 seconds but that wasn't the case. The tip for running an instant query worked perfectly for seeing what the raw data was!
